src/integer/mat_poly_over_z/sort.rs

```diff
@@ -0,0 +1,313 @@
+    ///         sum = sum + matrix.get_entry(row, 0)?.get_coeff(0)?;
+    ///     }
+    ///     Ok(sum)
+    /// }
+    ///
+    /// let sorted = mat.sort_by_column(custom_cond_func).unwrap();
+    ///
+    /// assert_eq!(cmp, sorted);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of the same type as `cond_func` if the execution of `cond_func` fails.
+    pub fn sort_by_column<T: Ord>(
+        &self,
+        cond_func: fn(&Self) -> Result<T, MathError>,
+    ) -> Result<Self, MathError> {
+        let mut condition_values = vec![];
+        for col in 0..self.get_num_columns() {
+            condition_values.push(cond_func(&self.get_column(col).unwrap())?);
+        }
+
+        let mut id_vec: Vec<usize> = (0..self.get_num_columns() as usize).collect();
+        id_vec.sort_by_key(|x| &condition_values[*x]);
+
+        let mut out = Self::new(self.get_num_rows(), self.get_num_columns());
+        for (col, item) in id_vec.iter().enumerate() {
+            out.set_column(col, self, *item).unwrap();
+        }
+
+        Ok(out)
+    }
+
+    /// Sorts the rows of the matrix based on some condition defined by `cond_func` in an ascending order.
+    ///
+    /// This condition is usually a norm with the described input-output behaviour.
+    ///
+    /// Parameters:
+    /// - `cond_func`: computes values implementing [`Ord`] over the columns of the specified matrix.
+    ///     These values are then used to re-order / sort the columns of the matrix.
+    ///
+    /// Returns an empty `Ok` if the action could be performed successfully.
+    /// A [`MathError`] is returned if the execution of `cond_func` returned an error.
+    ///
+    /// # Examples
+    /// ## Use a build-in function as condition
+    /// ```
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use std::str::FromStr;
+    /// let mat = MatPolyOverZ::from_str("[[2  3 4],[1  2],[1  1]]").unwrap();
+    /// let cmp = MatPolyOverZ::from_str("[[1  1],[1  2],[2  3 4]]").unwrap();
+    ///
+    /// let sorted = mat.sort_by_row(MatPolyOverZ::norm_infty).unwrap();
+    ///
+    /// assert_eq!(cmp, sorted);
+    /// ```
+    /// ## Use a custom function as condition
+    /// This function needs to take a row vector as input and output a type implementing [`PartialOrd`]
+    /// ```
+    /// use qfall_math::{integer::{MatPolyOverZ, Z}, error::MathError, traits::{MatrixDimensions, MatrixGetEntry}};
+    /// use crate::qfall_math::traits::GetCoefficient;
+    /// use std::str::FromStr;
+    /// let mat = MatPolyOverZ::from_str("[[2  0 4],[1  2],[1  1]]").unwrap();
+    /// let cmp = MatPolyOverZ::from_str("[[2  0 4],[1  1],[1  2]]").unwrap();
+    ///
+    /// fn custom_cond_func(matrix: &MatPolyOverZ) -> Result<Z, MathError> {
+    ///     let mut sum = Z::ZERO;
+    ///     for col in 0..matrix.get_num_columns() {
+    ///         sum = sum + matrix.get_entry(0, col)?.get_coeff(0)?;
+    ///     }
```

Comment:
```suggestion
    ///     for entry in 0..matrix.get_entries_columnwise() {
    ///         sum += entry.get_coeff(0)?;
    ///     }
```
Another minor suggestion to promote the use of `AddAssign` and the new iterators for matrices.
Please don't apply this in the function itself. It can't be more efficient than the iterator right now.